src/BatteryStateOfChargeService.h

```diff
@@ -17,6 +17,14 @@ class BatteryStateOfChargeService : public I_BatteryStateOfChargeService
 private:
     // Convention is to have an _ after private members
     const double initialStateOfChargePercent_;
-
+    double AvgCurrent;
+    double initialCurrent;
+    double changeTime;
+    QTime prevTime;
+    QTime currentTime;
+    double AmpHours;
+    double Current;
```

Comment:
You label the current time with `current` which is good, descriptive and I understand right away. This variable `Current` could use a prefix too, I wouldn't suggest `current` like you used for  `currentTime` (for obvious reasons).

Likewise, because you used `prev` for `prevTime`, following the same naming for `initialCurrent` would be better.